### PR TITLE
HAL for Hamming norm added

### LINKED ISSUE #25474: Add Hamming distance to OpenCV HAL
### Describe the feature and motivation

-

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by savuor**
fixes #25474

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/core/src/norm.cpp [hunk 1]
```diff
@@ -52,6 +52,9 @@ static const uchar popCountTable4[] =
 
 int normHamming(const uchar* a, int n, int cellSize)
 {
+    int output;
+    CALL_HAL_RET(normHamming8u, cv_hal_normHamming8u, output, a, n, cellSize);
```
Comment (asmorkalov) [inline modules/core/src/norm.cpp:56]: It should be just CAL_HALL, if you do not use output status.

Comment (savuor) [reply modules/core/src/norm.cpp:56]: CALL_HAL does not work with functions returning some value, looks like CALL_HAL_RET was made specially for that case

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by asmorkalov** [comment]
@vpisarev @opencv-alalek Do you have any opinion?

**Pull request merged**